components/event-gateway/pkg/nats/authenticator.go

```diff
@@ -283,19 +283,22 @@ func (a *automateAuthenticator) checkToken(token string) error {
 }
 
 func (a *automateAuthenticator) tryAuthzV2(ctx context.Context, subjects []string) (err error, shouldTryV1 bool) {
-	resp, err := a.authzClient.IsAuthorized(ctx, &authz.IsAuthorizedReq{
+	resp, err := a.authzClient.ProjectsAuthorized(ctx, &authz.ProjectsAuthorizedReq{
```

Comment:
The fact that this code exists at all is a giant smell to me. It wasn't working at all with IAM V2.1 so this at least pushes it in the right direction.